### PR TITLE
DTSSTCI-921

### PR DESCRIPTION
**Pull request by tomxelliott**
### Change description ###
Set up scheduled to task to migrate and tidy up inactive cases in DSS Draft state

### JIRA link ###
https://tools.hmcts.net/jira/browse/DTSSTCI-921

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/SystemClearInactiveDssDraftCasesTaskTest.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    @BeforeEach
+    void setUp() {
+        user = new User(SYSTEM_UPDATE_AUTH_TOKEN, UserDetails.builder().build());
+        when(idamService.retrieveSystemUpdateUserDetails()).thenReturn(user);
+        when(authTokenGenerator.generate()).thenReturn(SERVICE_AUTHORIZATION);
+    }
+
+    @Test
+    void shouldTriggerSystemTriggerCompleteHearingOutcome() {
+        final CaseDetails caseDetails1 = mock(CaseDetails.class);
```
Comment (rajaram-nagarajan) [inline src/test/java/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/SystemClearInactiveDssDraftCasesTaskTest.java:72]: Can we look for the possibilities to move the repeated code (case details related setup) to a common method or beforeEach method?

File: src/test/java/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/SystemClearInactiveDssDraftCasesTaskTest.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            .thenReturn(caseDetailsList);
+
+        task.run();
+
+        verify(ccdUpdateService).submitEvent(TEST_CASE_ID, SYSTEM_CLEAR_INACTIVE_DSS_DRAFT_CASE, user, SERVICE_AUTHORIZATION);
+        verify(ccdUpdateService).submitEvent(2L, SYSTEM_CLEAR_INACTIVE_DSS_DRAFT_CASE, user, SERVICE_AUTHORIZATION);
+    }
+
+    @Test
+    void shouldHandleCcdManagementExceptionWhenTriggerSystemTriggerCompleteHearingOutcomeFails() {
```
Comment (oleon23) [inline src/test/java/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/SystemClearInactiveDssDraftCasesTaskTest.java:91]: Looks like the naming of this test should be updated

File: src/test/java/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/SystemClearInactiveDssDraftCasesTaskTest.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+            .when(ccdUpdateService).submitEvent(TEST_CASE_ID, SYSTEM_CLEAR_INACTIVE_DSS_DRAFT_CASE, user, SERVICE_AUTHORIZATION);
+
+        task.run();
+
+        verify(ccdUpdateService).submitEvent(TEST_CASE_ID, SYSTEM_CLEAR_INACTIVE_DSS_DRAFT_CASE, user, SERVICE_AUTHORIZATION);
+        verify(ccdUpdateService).submitEvent(2L, SYSTEM_CLEAR_INACTIVE_DSS_DRAFT_CASE, user, SERVICE_AUTHORIZATION);
+    }
+
+    @Test
+    void shouldHandleIllegalArgumentExceptionWhenTriggerSystemTriggerCompleteHearingOutcomeFails() {
```
Comment (oleon23) [inline src/test/java/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/SystemClearInactiveDssDraftCasesTaskTest.java:106]: Same here - name needs updating

File: src/test/java/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/SystemClearInactiveDssDraftCasesTaskTest.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        caseDetails1 = mock(CaseDetails.class);
+        caseDetails2 = mock(CaseDetails.class);
+        caseDetailsList = List.of(caseDetails1, caseDetails2);
+
+        when(idamService.retrieveSystemUpdateUserDetails()).thenReturn(user);
+        when(authTokenGenerator.generate()).thenReturn(SERVICE_AUTHORIZATION);
+    }
+
+    @Test
+    void shouldTriggerSystemTriggerCompleteHearingOutcome() {
```
Comment (oleon23) [inline src/test/java/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/SystemClearInactiveDssDraftCasesTaskTest.java:78]: Looks like the naming of this test should be updated

File: src/test/java/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/SystemClearInactiveDssDraftCasesTaskTest.java (hunk 2 above)
Comment (tomxelliott) [reply src/test/java/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/SystemClearInactiveDssDraftCasesTaskTest.java:91]: 👍

File: src/test/java/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/SystemClearInactiveDssDraftCasesTaskTest.java (hunk 4 above)
Comment (tomxelliott) [reply src/test/java/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/SystemClearInactiveDssDraftCasesTaskTest.java:78]: 👍

**Review by oleon23 — APPROVED** [review]

**Pull request merged**